src/Centreon/Application/Controller/Monitoring/CommentController.php

```diff
@@ -0,0 +1,346 @@
+            Constraint::CHECK_MODE_VALIDATE_SCHEMA
+        );
+
+        if (!$validator->isValid()) {
+            $message = '';
+            foreach ($validator->getErrors() as $error) {
+                $message .= sprintf("[%s] %s\n", $error['property'], $error['message']);
+            }
+            throw new \InvalidArgumentException($message);
+        }
+
+        return $results;
+    }
+
+    /**
+     * This function will verify that the contact is authorized to add a comment
+     * on the selected resources
+     *
+     * @param Contact $contact
+     * @param array $resources
+     * @return boolean
+     */
+    private function hasCommentRightsForResources(Contact $contact, array $resources): bool
+    {
+        if ($contact->isAdmin()) {
+            return true;
+        }
+        /**
+         * Retrieving the current rights of the user for adding comments
+         */
+        $hasHostRights = $contact->hasRole(Contact::ROLE_HOST_ADD_COMMENT);
+        $hasServiceRights = $contact->hasRole(Contact::ROLE_SERVICE_ADD_COMMENT);
+
+        /**
+         * If the user has no rights at all, do not go further
+         */
+        if (!$hasHostRights && !$hasServiceRights) {
+            return false;
+        }
+
+        foreach ($resources as $resource) {
+            if (
+                ($resource['type'] === ResourceEntity::TYPE_HOST && $hasHostRights)
+                || ($resource['type'] === ResourceEntity::TYPE_SERVICE && $hasServiceRights)
+            ) {
+                continue;
+            }
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Entry point to add comments to multiple resources
+     *
+     * @param Request $request
+     * @return View
+     * @throws \Exception
+     * @throws \InvalidArgumentException
+     */
+    public function addResourcesComment(
+        Request $request
+    ): View {
+        $this->denyAccessUnlessGrantedForApiRealtime();
+        /**
+        * @var Contact $contact
+        */
+        $contact = $this->getUser();
+        $this->commentService->filterByContact($contact);
+
+       /*
+        * Validate the content of the POST request against the JSON schema validator
+        */
+        $results = $this->validateAndRetrievePostData(
+            $request,
+            'config/json_validator/latest/Centreon/Comment/CommentResources.json'
+        );
+
+        /**
+         * If user has no rights to add a comment for host and/or service
+         * return view with unauthorized HTTP header response
+         */
+        if (!$this->hasCommentRightsForResources($contact, $results['resources'])) {
+            return $this->view(null, Response::HTTP_UNAUTHORIZED);
+        }
+
+        /**
+         * Dissect the results to extract the hostids and serviceids from it
+         */
+        $hostIds = [];
+        $serviceIds = [];
+        $comments = [];
+
+        $now = new \DateTime('now');
```

Comment:
```suggestion
        $now = new \DateTime();
```
'now' is the default value for a new \DateTime()